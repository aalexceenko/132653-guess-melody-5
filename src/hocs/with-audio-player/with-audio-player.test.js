import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withActivePlayer from "./with-audio-player";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withActivePlayer(MockComponent);

it(`withActivePlayer is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped>
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
