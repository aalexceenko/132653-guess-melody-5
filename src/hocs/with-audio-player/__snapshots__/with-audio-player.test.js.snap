// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActivePlayer is rendered correctly 1`] = `<div />`;
